android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java

```diff
@@ -1,41 +1,51 @@
 package com.globo.reactnativeua;
 
+import android.app.ActivityManager;
 import android.content.Context;
+import android.content.Intent;
 import android.support.annotation.NonNull;
 
 import com.urbanairship.AirshipReceiver;
 import com.urbanairship.push.PushMessage;
 
+import java.util.List;
+
 
 public class ReactNativeUAReceiver extends AirshipReceiver {
 
     @Override
     protected void onPushReceived(@NonNull Context context, @NonNull PushMessage message, boolean notificationPosted) {
-        if (ReactNativeUAEventEmitter.getInstance() != null) {
-            ReactNativeUAEventEmitter.getInstance().sendEvent("onPushReceived", message);
-        }
-    }
+        boolean isRunning = isApplicationRunning(context);
 
-    @Override
-    protected void onNotificationPosted(@NonNull Context context, @NonNull NotificationInfo notificationInfo) {
-        if (ReactNativeUAEventEmitter.getInstance() != null) {
-            ReactNativeUAEventEmitter.getInstance().sendEvent("onNotificationPosted", notificationInfo.getMessage());
+        if (isRunning && ReactNativeUAEventEmitter.getInstance() != null) {
+            ReactNativeUAEventEmitter.getInstance().sendEvent("onNotificationOpened", message);
```

Comment:
send event to react-native if the emitter is initialized and application is running